### PR TITLE
fix(functional-tests): report unknown terminal state 'system-err'

### PR DESCRIPTION
**Pull request by Trinaa**
## Because

- test failures are being suppressed and are reported with status 'unknown' in CircleCI

## This pull request

- removes instances of 'console.error' which causes the invalid status
- adds basic ESLint to functional-tests to prevent additions of 'console.error'

## Issue that this pull request solves

Closes: # [FXA-8773](https://mozilla-hub.atlassian.net/browse/FXA-8773)

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [x] I have added necessary documentation (if appropriate).
- [x] I have verified that my changes render correctly in RTL (if appropriate).

## Screenshots (Optional)

Please attach the screenshots of the changes made in case of change in user interface.

## Other information (Optional)

Any other information that is important to this pull request.

- [Lint failure example](https://app.circleci.com/pipelines/github/mozilla/fxa/46792/workflows/508ec67a-2dac-42ea-ac39-3dab2799acff/jobs/477840?invite=true#step-103-16393_56)
- [Test failure example](https://app.circleci.com/pipelines/github/mozilla/fxa/46793/workflows/bceeb156-513d-43fb-aa13-97ce856e6d38/jobs/477856/tests)

Co-authored-by: @pdehaan 

[FXA-8773]: https://mozilla-hub.atlassian.net/browse/FXA-8773?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by pdehaan — APPROVED** [review]
:+1: Let's land this and see how it works in main and Nightly CI runs for a few weeks.

**Review by ashrivastava-qa — COMMENTED** [review]
Just have a question that I added above. Rest everything looks fine.

File: packages/functional-tests/tests/oauth/oauthPermissions.spec.ts [hunk 1]
```diff
@@ -18,14 +18,7 @@ test.describe('severity-1 #smoke', () => {
...
         // Cleanup any accounts created during the test
-        try {
-          await target.auth.accountDestroy(email, password);
-        } catch (e) {
-          // Handle the error here
-          console.error('An error occurred during account cleanup:', e);
-          // Optionally, rethrow the error to propagate it further
-          throw e;
-        }
+        await target.auth.accountDestroy(email, password);
```
[the remark this answers is not in the document]
Comment (Trinaa) [reply packages/functional-tests/tests/oauth/oauthPermissions.spec.ts:21]: The test will fail if the cleanup fails, there isn't an explicit message that the failure occurred in the cleanup, but there is a stack trace that seems easy enough to reconcile. 

I ran an experiment to see the outcome [here](https://app.circleci.com/pipelines/github/mozilla/fxa/46793/workflows/bceeb156-513d-43fb-aa13-97ce856e6d38/jobs/477856/tests). So if we take the `severity-1 #smoke › signup react › signup web` test for example, it says it failed on `react-conversion/signup.spec.ts:98:9` which is in the cleanup code.

[the remark this answers is not in the document]
Comment (ashrivastava-qa) [reply packages/functional-tests/tests/oauth/oauthPermissions.spec.ts:21]: For now its good, but I would like to have more context into the failure cause/log. It makes the debugging easier which has been a challenge in the past.

**Pull request merged**